executor/linux/build.go

```diff
@@ -42,6 +44,76 @@ func (c *client) CreateBuild(ctx context.Context) error {
 		return fmt.Errorf("unable to upload build state: %w", c.err)
 	}
 
+	// before setting up the build, enforce repo.trusted is set for pipelines containing privileged images
+	// this configuration is set as an executor flag
+	if c.enforceTrustedRepos {
+		// check if pipeline steps contain privileged images
+		// assume no privileged images are in use
+		containsPrivilegedImages := false
+
+		// group steps services and stages together
+		containers := c.pipeline.Steps
+
+		containers = append(containers, c.pipeline.Services...)
+		for _, stage := range c.pipeline.Stages {
+			containers = append(containers, stage.Steps...)
+		}
+
+		for _, container := range containers {
+			// TODO: remove hardcoded reference
+			//
+			//nolint:goconst // ignore making a constant for now
+			if container.Name == "init" {
```

Comment:
Would the following config be exempt from the `containsPrivilegedImages` check, or does `container.Name` != the `name` key from the config?

```yaml
---
version: "1"

steps:
- name: init
  image: docker.io/library/alpine:3.16
  commands:
  - whoami
```

Reply:
good catch @BrandonIngalls. so that pipeline you pasted wouldnt work because of this line:
https://github.com/go-vela/worker/blob/main/executor/linux/step.go#L31-L33
which skips any steps and stages with "init" as the name from being created.

however a service named "init" would still get through so just to be safe i changed the code to skip over the hardcoded init image instead:
https://github.com/go-vela/worker/commit/784719552c2527a7ac6163716a80026a1ee82da6
https://github.com/go-vela/server/blob/main/compiler/native/initialize.go#L14

that should work better